src/commons/Post.tsx

```diff
@@ -0,0 +1,347 @@
+import { TouchableOpacity } from 'react-native-gesture-handler';
+import AccessLevelAvatar from './AccessLevelAvatar';
+import HTMLRenderer from './HTMLRenderer';
+import { like, likeOn, menuHSvg, replies } from '../assets/svgs';
+import { likePost, disLikePost, connection } from '../services/forum.service';
+import { updatePosts } from '../redux/threads/ThreadActions';
+import { IS_TABLET } from '../services/helpers';
+import { useDispatch } from 'react-redux';
+import { useNavigation } from '@react-navigation/native';
+import type { IAuthor, IPost, IUser } from '../entity/IForum';
+import { useAppSelector } from '../redux/Store';
+import type { StackNavigationProp } from '@react-navigation/stack';
+import type { ForumRootStackParamList } from '../entity/IRouteParams';
+
+interface IPostProps {
+  id: number;
+  appColor: string;
+  index: number;
+  isDark: boolean;
+  locked: boolean;
+  user?: IUser;
+  selected: boolean;
+  toggleMenu: (post: IPost, type: 'user' | 'post') => void;
+  onPostCreated: (postId?: number) => void;
+}
+
+const Post: FunctionComponent<IPostProps> = props => {
+  const { id, appColor, index, isDark, locked, user, selected, toggleMenu, onPostCreated } = props;
+  const { navigate } = useNavigation<StackNavigationProp<ForumRootStackParamList>>();
+  const dispatch = useDispatch();
+  const styles = setStyles(isDark, appColor);
+  const signShown = useAppSelector(state => state.threadsState?.signShown);
+  const post = useAppSelector(state => state?.threadsState?.posts?.[id]);
+  const [isLiked, setIsLiked] = useState(post?.is_liked_by_viewer);
+  const [likeCount, setLikeCount] = useState(post?.like_count || 0);
+
+  const baseColor = isDark ? '#081825' : '#FFFFFF';
+  const selectedColor = isDark ? '#002039' : '#E1E6EB';
+
+  const toggleLike = useCallback(() => {
+    if (!connection(true)) {
+      return;
+    }
+    if (post) {
+      if (isLiked) {
+        setLikeCount(likeCount - 1);
+        disLikePost(post?.id);
+      } else {
+        setLikeCount(likeCount + 1);
+        likePost(post?.id);
+      }
+      setIsLiked(!isLiked);
+      dispatch(
+        updatePosts({
+          ...post,
+          is_liked_by_viewer: !isLiked,
+          like_count: isLiked ? likeCount - 1 : likeCount + 1,
+        })
+      );
+    }
+  }, [isLiked, likeCount, post, dispatch]);
+
+  const onNavigateToCoach = useCallback(
+    (coachId: number) => {
+      navigate('CoachOverview', { coachId });
+    },
+    [navigate]
+  );
+
+  const reply = useCallback(() => {
+    if (post) {
+      navigate('CRUD', {
+        type: 'post',
+        action: 'create',
+        threadId: post?.thread_id,
+        onPostCreated,
+        quotes: [
+          {
+            ...post,
+            content: `<blockquote><b>${post?.author?.display_name}</b>:<br>${post?.content}</blockquote>`,
+          },
+        ],
+      });
+    }
+  }, [navigate, onPostCreated, post]);
+
+  return (

```

Comment:
Something is off with how post are displayed when a user replies to a post or multiquotes multiple quotes, see images (1st is from your branch, 2nd from master)
![image](https://github.com/railroadmedia/react-native-musora-forum/assets/32734435/7ebdd0b3-7cd5-421f-ab04-2ebb5c70e390)
![image](https://github.com/railroadmedia/react-native-musora-forum/assets/32734435/60336e9b-94fe-4432-b43f-6d03b62db25f)

Reply:
Fixed it.